tests/foreman/cli/test_discoveredhost.py

```diff
@@ -15,13 +15,131 @@
 
 @Upstream: No
 """
-from robottelo.decorators import run_only_on, stubbed, tier3
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.discoveryhost import DiscoveryHost
+from robottelo.cli.factory import make_org, make_location
+from robottelo.cli.settings import Settings
+from robottelo.cli.template import Template
+from robottelo.decorators import run_only_on, skip_if_not_set, stubbed, tier3
+from robottelo.libvirt_discovery import LibvirtGuest
 from robottelo.test import CLITestCase
+from time import sleep
 
 
 class DiscoveryTestCase(CLITestCase):
     """Implements Foreman discovery CLI tests."""
 
+    def _assertdiscoveredhost(self, hostname):
+        """Check if host is discovered and information about it can be
+        retrieved back
+
+        Introduced a delay of 300secs by polling every 10 secs to get expected
+        host
+        """
+        for _ in range(30):
```

Comment:
Suggesting a small improvement here - to accept optional `timeout=` parameter with default value of 300. And updating thing loop to `for _ in range(timeout / 10):`. This way we can easily update the timeout later for tests which are faster/slower than these initial ones.